### PR TITLE
print full path to history file on parse error

### LINKED ISSUE #18223: print full path to history file when corrupted
https://groups.google.com/forum/#!topic/julia-users/IK4fIezx0Uc

### PR DESCRIPTION
**Pull request by KristofferC**
fixes #18223

### CODE REVIEW

**Comment by KristofferC** [comment]
Comments?

**Comment by tkelman** [comment]
Looks reasonable. OSX Travis log backed up to https://gist.github.com/a953a4f97617e5748824334d265604b5 and restarted, I fixed that problem in #18241

**Comment by KristofferC** [comment]
Ok to merge?

File: base/REPL.jl [hunk 1]
```diff
@@ -771,9 +771,10 @@ function setup_interface(repl::LineEditREPL; hascolor = repl.hascolor, extra_rep
                                               :help  => help_mode))
     if repl.history_file
         try
-            f = open(find_hist_file(), true, true, true, false, false)
+            hist_path = find_hist_file()
+            f = open(hist_path, true, true, true, false, false)
```
Comment (tkelman) [inline base/REPL.jl]: should this get closed a little sooner?

Comment (KristofferC) [inline base/REPL.jl]: Not sure what you mean. It gets closed when the `replc` gets garbage collected. When do you want it to be closed?

**Comment by tkelman** [comment]
leaving for 0.5.1

**Comment by tkelman** [comment]
At least one package was using `hist_from_file` so I think it'll need a fallback 2-argument method for backporting.

**Pull request merged**